cmd/skopeo/utils_test.go

```diff
@@ -180,3 +181,50 @@ func TestImageDestOptionsNewSystemContext(t *testing.T) {
 	_, err = opts.newSystemContext()
 	assert.Error(t, err)
 }
+
+// since there is a shared authfile image option and a non-shared (prefixed) one, make sure the override logic
+// works correctly.
+func TestImageOptionsAuthfileOverride(t *testing.T) {
+	// if authfile and dest-authfile is provided, dest-authfile wins
```

Comment:
This would be much shorter, and more readable, as a table-driven test (see `TestImageOptionsNewSystemContext` for an example).

It would also be nice to add a test for the `prefix == ""` case.

Reply:
refactored and included empty prefix case.